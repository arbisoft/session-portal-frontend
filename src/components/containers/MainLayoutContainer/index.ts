import MainLayoutContainer from "./mainLayoutContainer";
export default MainLayoutContainer;
