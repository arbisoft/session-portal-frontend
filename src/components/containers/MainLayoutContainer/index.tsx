"use client";

import React from "react";
import Container from "@mui/material/Container";
import Box from "@mui/material/Box";
import Stack from "@mui/material/Stack";
import { usePathname } from "next/navigation";

type TMainLayoutContainer = {
  children: React.ReactNode;
};

const MainLayoutContainer = ({ children }: TMainLayoutContainer) => {
  const pathname = usePathname();
  const showRightBar = pathname.includes("/watch");

  return (
    <Container sx={{ height: "100svh" }}>
      <Stack flexDirection="row" height="100%">
        <Box height="100%" width="220px" border="1px solid #ddd"></Box>

        <Box flexGrow={1} height="100%" padding="2px">
          {children}
        </Box>

        {showRightBar && (
          <Box height="100%" width="230px" border="1px solid #ddd">
            <h3>Suggested for You</h3>
          </Box>
        )}
      </Stack>
    </Container>
  );
};

export default MainLayoutContainer;
