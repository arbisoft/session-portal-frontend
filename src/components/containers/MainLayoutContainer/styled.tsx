import { backdropClasses } from "@mui/material/Backdrop";
import Box from "@mui/material/Box";
import Container from "@mui/material/Container";
import Drawer, { drawerClasses } from "@mui/material/Drawer";
import Grid2 from "@mui/material/Grid2";
import { styled, css } from "@mui/material/styles";

export const MainContainer = styled(Container, {
  name: "MainContainer",
})(({ theme }) => {
  return css`
    display: flex;
    gap: 30px;
    min-height: 100vh;
    padding: ${theme.spacing(4, 0)};
    padding-top: ${theme.spacing(12)};
  `;
});

export const LeftSidebar = styled(Box, {
  name: "LeftSidebar",
})(() => {
  return css`
    flex-shrink: 0;
    width: 240px;
  `;
});

export const ContentContainer = styled(Grid2, {
  name: "ContentContainer",
})(({ theme }) => {
  return css`
    align-content: flex-start;
    flex: 1;
    ${theme.breakpoints.down("sm")} {
      padding: ${theme.spacing(0, 2)};
    }
  `;
});

export const RightSidebar = styled(Box, {
  name: "RightSidebar",
})(() => {
  return css`
    flex-shrink: 0;
    width: 300px;
  `;
});

export const StyledDrawer = styled(Drawer, {
  name: "StyledDrawer",
})(({ theme }) => {
  return css`
    &,
    .${backdropClasses.root}, .${drawerClasses.paper} {
      top: 64px;

      ${theme.breakpoints.down("lg")} {
        top: 56px;
      }
    }
    .${backdropClasses.root} {
      background-color: transparent;
    }
  `;
});
