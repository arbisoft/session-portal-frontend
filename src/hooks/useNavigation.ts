import { useRouter } from "next/navigation";

export const ALLOWED_PAGES = {
  get home(): string {
    return "/";
  },
  get login(): string {
    return "/login";
  },
  get videos(): string {
    return "/videos";
  },
  get searchResult(): string {
    return `${this.videos}/results`;
  },
  videoDetail(id: string) {
    return `/videos/${id}`;
  },
  get uploadVideo(): string {
    return "/upload-video";
  },
} as const;

type Page = keyof typeof ALLOWED_PAGES;

const useNavigation = () => {
  const router = useRouter();

  /**
   * Generate a dynamic URL based on the page and optional params.
   * @param page - The target page key from ALLOWED_PAGES.
   * @param params - Query parameters or required dynamic ID (for `videoDetail`).
   * @returns Fully constructed localized URL.
   */
  const getPageUrl = (page: Page, params?: Record<string, string | number>) => {
    let pageUrl = ALLOWED_PAGES[page];

    if (typeof pageUrl === "function" && params?.id) {
      pageUrl = pageUrl(String(params.id));
      delete params.id;
    }

    const queryString =
      params && Object.keys(params).length ? `?${new URLSearchParams(params as Record<string, string>).toString()}` : "";

    return `${pageUrl}${queryString}`;
  };

  /**
   * Navigate to a new page dynamically.
   * @param page - The target page.
   * @param params - Query parameters or required `id` for `videoDetail`.
   */
  const navigateTo = (page: Page, params?: Record<string, string | number>) => {
    const url = getPageUrl(page, params);
    router.push(url, { scroll: true });
  };

  return { getPageUrl, navigateTo };
};

export default useNavigation;
