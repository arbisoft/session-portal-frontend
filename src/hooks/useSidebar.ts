import { useEffect, useState } from "react";

const useSidebar = () => {
  const [sidebarItems, setSidebarItems] = useState<string[]>([]);

  useEffect(() => {
    const onGetSidebarItems = async () => {
      //   TODO: Remove static code and get items from backend
      setSidebarItems(["All", "All-Hands Meetings", "AMA Sessions", "Product Showcase", "Nurture Sessions", "Security Session"]);
    };
    onGetSidebarItems();
  }, []);

  return { sidebarItems };
};
export default useSidebar;
