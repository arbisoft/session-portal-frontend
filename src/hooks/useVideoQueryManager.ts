import { useEffect, useMemo, useState } from "react";

import { startOfYear, endOfYear, format } from "date-fns";

import { defaultParams } from "@/features/VideosListingPage/types";
import { EventsParams } from "@/models/Events";
import { parseNonPassedParams } from "@/utils/utils";

export function useVideoQueryManager(searchParams: URLSearchParams) {
  const [page, setPage] = useState(1);

  const parsedParams = useMemo(
    () => ({
      tag: searchParams.get("tag") || null,
      playlist: searchParams.get("playlist") || null,
      search: searchParams.get("search") || null,
      order: searchParams.get("order") || null,
      year: searchParams.get("year") || null,
    }),
    [searchParams]
  );

  useEffect(() => {
    setPage(1);
  }, [parsedParams.tag, parsedParams.playlist, parsedParams.search, parsedParams.order, parsedParams.year]);

  const apiParams = useMemo(() => {
    return parseNonPassedParams({
      ...defaultParams,
      page,
      page_size: 12,
      ordering: parsedParams.order ? [parsedParams.order] : ["-event_time"],
      search: parsedParams.search ?? undefined,
      tag: parsedParams.search ? "" : (parsedParams.tag ?? ""),
      playlist: parsedParams.search ? "" : (parsedParams.playlist ?? ""),
      event_time_after: parsedParams.year ? format(startOfYear(new Date(parsedParams.year)), "yyyy-MM-dd") : undefined,
      event_time_before: parsedParams.year ? format(endOfYear(new Date(parsedParams.year)), "yyyy-MM-dd") : undefined,
    }) as EventsParams;
  }, [page, parsedParams]);

  return {
    page,
    setPage,
    parsedParams,
    apiParams,
  };
}
