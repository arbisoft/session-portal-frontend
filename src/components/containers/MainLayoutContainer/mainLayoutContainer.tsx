"use client";

import React from "react";

import { usePathname } from "next/navigation";

import { MainContainer, MainStack, SidebarContainer, RightBarContainer, ContentContainer } from "./styled";

type TMainLayoutContainer = {
  children: React.ReactNode;
};

const MainLayoutContainer = ({ children }: TMainLayoutContainer) => {
  const pathname = usePathname();
  const showRightBar = pathname.includes("/");

  return (
    <MainContainer>
      <MainStack>
        <SidebarContainer></SidebarContainer>

        <ContentContainer>{children}</ContentContainer>

        {showRightBar && (
          <RightBarContainer>
            <h3>Suggested for You</h3>
          </RightBarContainer>
        )}
      </MainStack>
    </MainContainer>
  );
};

export default MainLayoutContainer;
