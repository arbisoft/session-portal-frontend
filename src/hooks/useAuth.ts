import { useEffect } from "react";

import { usePathname } from "next/navigation";
import { useSelector } from "react-redux";

import { selectAccessToken } from "@/redux/login/selectors";

import useNavigation from "./useNavigation";

const useAuth = () => {
  const pathname = usePathname();
  const token = useSelector(selectAccessToken);
  const { navigateTo, getPageUrl } = useNavigation();

  useEffect(() => {
    const isRootOrLoginPage = getPageUrl("home").includes(pathname) || pathname === getPageUrl("login");

    if (token && isRootOrLoginPage) {
      navigateTo("videos");
    } else if (!token) {
      navigateTo("login");
    }
  }, [token, pathname]);
};

export default useAuth;
