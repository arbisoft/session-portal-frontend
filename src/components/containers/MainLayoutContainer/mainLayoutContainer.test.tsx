import { useSearchParams } from "next/navigation";

import { useFeatureFlags } from "@/hooks/useFeatureFlags";
import { act, fireEvent, customRender as render, screen, waitFor } from "@/jest/utils/testUtils";

import MainLayoutContainer from "./mainLayoutContainer";

jest.mock("next/navigation", () => ({
  ...jest.requireActual("next/navigation"),
  useSearchParams: jest.fn(),
  usePathname: jest.fn().mockReturnValue("/some-path"),
}));

const mockNavigateTo = jest.fn();
jest.mock("@/hooks/useNavigation", () => ({
  __esModule: true,
  default: () => ({
    navigateTo: mockNavigateTo,
    getPageUrl: jest.fn((key) => key),
  }),
}));

const mockItems = [
  { id: "1", name: "Test-1" },
  { id: "2", name: "Test-2" },
  { id: "3", name: "Test-3" },
];

jest.mock("@/hooks/useSidebar", () => ({
  __esModule: true,
  default: jest.fn(() => ({
    playlists: mockItems,
    arePlaylistsLoading: false,
    tags: mockItems,
  })),
}));

describe("MainLayoutContainer", () => {
  beforeEach(() => {
    (useSearchParams as jest.Mock).mockReturnValue({
      get: jest.fn().mockReturnValue("test search"),
    });

    (useFeatureFlags as jest.Mock).mockReturnValue({
      isFeatureEnabled: (feature: string) => feature === "test",
    });

    jest.clearAllMocks();
  });

  test("should renders Navbar component", async () => {
    await waitFor(async () => {
      render(
        <MainLayoutContainer>
          <div>Test Content</div>
        </MainLayoutContainer>
      );
    });

    expect(screen.getByTestId("navbar")).toBeInTheDocument();
  });

  test("should RightSidebar should be hidden on small screens (xs)", () => {
    global.innerWidth = 500;
    global.dispatchEvent(new Event("resize"));

    render(
      <MainLayoutContainer rightSidebar={<div>Right Content</div>}>
        <div>Test Content</div>
      </MainLayoutContainer>
    );

    expect(globalThis.innerWidth).toBe(500);

    waitFor(() => {
      expect(screen.queryByTestId("right-sidebar")).toBeNull();
    });
  });

  test("should RightSidebar should be visible on medium and larger screens (md+)", () => {
    global.innerWidth = 1024;
    window.dispatchEvent(new Event("resize"));

    render(
      <MainLayoutContainer rightSidebar={<div>Right Content</div>}>
        <div>Test Content</div>
      </MainLayoutContainer>
    );

    waitFor(() => {
      expect(screen.queryByTestId("right-sidebar")).toHaveStyle("display:block");
    });
  });

  test("should renders Sidebar component inside LeftSidebar", () => {
    render(
      <MainLayoutContainer>
        <div>Test Content</div>
      </MainLayoutContainer>
    );

    expect(screen.getByTestId("left-sidebar")).toBeInTheDocument();
  });

  test("should renders children inside ContentContainer", () => {
    render(
      <MainLayoutContainer>
        <div data-testid="content">Test Content</div>
      </MainLayoutContainer>
    );

    expect(screen.getByTestId("content")).toBeInTheDocument();
  });

  test("should does not render RightSidebar if rightSidebar prop is undefined", () => {
    render(
      <MainLayoutContainer>
        <div>Test Content</div>
      </MainLayoutContainer>
    );

    expect(screen.queryByTestId("right-sidebar")).not.toBeInTheDocument();
  });

  test("should renders RightSidebar when rightSidebar prop is provided", () => {
    render(
      <MainLayoutContainer rightSidebar={<div>Right Content</div>}>
        <div>Test Content</div>
      </MainLayoutContainer>
    );

    expect(screen.getByTestId("right-sidebar")).toBeInTheDocument();
  });

  test("should renders multiple children correctly", () => {
    render(
      <MainLayoutContainer>
        <div data-testid="child-1">Child 1</div>
        <div data-testid="child-2">Child 2</div>
      </MainLayoutContainer>
    );

    expect(screen.getByTestId("child-1")).toBeInTheDocument();
    expect(screen.getByTestId("child-2")).toBeInTheDocument();
  });

  test("should matches snapshot for UI consistency", () => {
    const { asFragment } = render(
      <MainLayoutContainer rightSidebar={<div>Right Content</div>}>
        <div>Test Content</div>
      </MainLayoutContainer>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test("should handle drawer open and close correctly when shouldShowDrawer is true", async () => {
    render(
      <MainLayoutContainer shouldShowDrawer>
        <div>Test Content</div>
      </MainLayoutContainer>
    );

    const menuButton = screen.getByTestId("open-drawer");
    act(() => {
      fireEvent.click(menuButton);
    });

    await waitFor(() => {
      expect(screen.getByTestId("drawer")).toBeInTheDocument();
    });

    act(() => {
      fireEvent.click(screen.getByTestId("drawer"));
    });

    await waitFor(() => {
      expect(screen.getByTestId("drawer").firstChild).toHaveAttribute("aria-hidden", "true");
    });
  });
});
