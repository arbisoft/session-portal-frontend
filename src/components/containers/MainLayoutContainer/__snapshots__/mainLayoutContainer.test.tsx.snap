// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainLayoutContainer should matches snapshot for UI consistency 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-m8dnq0"
  >
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-b4hu5c-MuiPaper-root-MuiAppBar-root-StyledAppBar-root"
      data-testid="navbar"
      style="--Paper-shadow: var(--mui-shadows-4); --Paper-overlay: var(--mui-overlays-4);"
    >
      <div
        class="MuiContainer-root css-12f197v-MuiContainer-root"
      >
        <div
          class="MuiToolbar-root MuiToolbar-regular css-1u9o2lj-MuiToolbar-root"
        >
          <a
            class="css-jiztmo-Logo-root"
            data-testid="navbar-logo"
            href="/videos"
          >
            <img
              alt="Arbisoft Icon"
              data-nimg="1"
              data-testid="arbisoftLogo"
              decoding="async"
              height="24"
              loading="lazy"
              src="/_next/image?url=%2Fassets%2Fimages%2Fapple-icon.png&w=48&q=75"
              srcset="/_next/image?url=%2Fassets%2Fimages%2Fapple-icon.png&w=32&q=75 1x, /_next/image?url=%2Fassets%2Fimages%2Fapple-icon.png&w=48&q=75 2x"
              style="color: transparent;"
              width="24"
            />
            <h6
              class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-1k6p2lw-MuiTypography-root"
            >
              Arbisoft Sessions Portal
            </h6>
          </a>
          <div
            class="MuiBox-root css-1rqqp05"
          >
            <form
              class="css-asd7gd-Search-root"
            >
              <div
                class="MuiInputBase-root MuiInputBase-colorPrimary css-1mhii58-MuiInputBase-root-StyledInputBase-root"
              >
                <input
                  aria-label="search"
                  class="MuiInputBase-input css-1943rd7-MuiInputBase-input"
                  data-testid="search-query"
                  placeholder="Search..."
                  type="text"
                  value="test search"
                />
              </div>
              <button
                aria-label="cancel"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1bf1yfo-MuiButtonBase-root-MuiIconButton-root"
                data-testid="cancelIcon"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-4uxqju-MuiSvgIcon-root"
                  data-testid="CancelIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2m5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12z"
                  />
                </svg>
              </button>
              <button
                aria-label="search"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-ndsf0k-MuiButtonBase-root-MuiIconButton-root-SearchIconWrapper-root"
                tabindex="0"
                type="submit"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-4uxqju-MuiSvgIcon-root"
                  data-testid="SearchIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14"
                  />
                </svg>
              </button>
            </form>
          </div>
          <div
            class="MuiBox-root css-si7qf5-NavbarRightArea-root"
          >
            <button
              aria-label="Open settings"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-9nlk2f-MuiButtonBase-root-MuiIconButton-root"
              data-mui-internal-clone-element="true"
              data-testid="avatar-btn"
              tabindex="0"
              type="button"
            >
              <div
                class="MuiAvatar-root MuiAvatar-rounded MuiAvatar-colorDefault css-5ta4e5-MuiAvatar-root"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-pamvpt-MuiSvgIcon-root-MuiAvatar-fallback"
                  data-testid="PersonIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
                  />
                </svg>
              </div>
            </button>
          </div>
        </div>
      </div>
    </header>
    <div
      class="MuiBox-root css-1ylnd6d"
      data-testid="left-sidebar"
    >
      <nav
        aria-label="Video navigation sidebar"
        class="css-4kypo5"
        data-testid="sidebar-container"
      >
        <div
          class="MuiStack-root css-ujfjh4-MuiStack-root-MenuStack-root"
        >
          <ul
            aria-label="Video playlists"
            class="MuiList-root MuiList-padding css-18jbpcj-MuiList-root"
            role="menu"
            tabindex="-1"
          >
            <a
              class="MuiButtonBase-root MuiMenuItem-root MuiMenuItem-gutters MuiMenuItem-root MuiMenuItem-gutters css-19f1w6o-MuiButtonBase-root-MuiMenuItem-root"
              data-testid="sidebar-item-All"
              href="/videos"
              role="menuitem"
              tabindex="0"
            >
              <img
                alt="sidebar item icon"
                aria-hidden="true"
                data-nimg="1"
                decoding="async"
                height="12"
                loading="lazy"
                src="/assets/images/sidebar-item-icon.svg"
                style="color: transparent;"
                width="18"
              />
              <p
                class="MuiTypography-root MuiTypography-bodySmall css-fmwgzn-MuiTypography-root"
                title="All videos"
              >
                All videos
              </p>
            </a>
            <a
              class="MuiButtonBase-root MuiMenuItem-root MuiMenuItem-gutters MuiMenuItem-root MuiMenuItem-gutters css-19f1w6o-MuiButtonBase-root-MuiMenuItem-root"
              data-testid="sidebar-item-Test-1"
              href="/videos?playlist=Test-1"
              role="menuitem"
              tabindex="-1"
            >
              <img
                alt="sidebar item icon"
                aria-hidden="true"
                data-nimg="1"
                decoding="async"
                height="12"
                loading="lazy"
                src="/assets/images/sidebar-item-icon.svg"
                style="color: transparent;"
                width="18"
              />
              <p
                class="MuiTypography-root MuiTypography-bodySmall css-fmwgzn-MuiTypography-root"
                title="Test-1"
              >
                Test-1
              </p>
            </a>
            <a
              class="MuiButtonBase-root MuiMenuItem-root MuiMenuItem-gutters MuiMenuItem-root MuiMenuItem-gutters css-19f1w6o-MuiButtonBase-root-MuiMenuItem-root"
              data-testid="sidebar-item-Test-2"
              href="/videos?playlist=Test-2"
              role="menuitem"
              tabindex="-1"
            >
              <img
                alt="sidebar item icon"
                aria-hidden="true"
                data-nimg="1"
                decoding="async"
                height="12"
                loading="lazy"
                src="/assets/images/sidebar-item-icon.svg"
                style="color: transparent;"
                width="18"
              />
              <p
                class="MuiTypography-root MuiTypography-bodySmall css-fmwgzn-MuiTypography-root"
                title="Test-2"
              >
                Test-2
              </p>
            </a>
            <a
              class="MuiButtonBase-root MuiMenuItem-root MuiMenuItem-gutters MuiMenuItem-root MuiMenuItem-gutters css-19f1w6o-MuiButtonBase-root-MuiMenuItem-root"
              data-testid="sidebar-item-Test-3"
              href="/videos?playlist=Test-3"
              role="menuitem"
              tabindex="-1"
            >
              <img
                alt="sidebar item icon"
                aria-hidden="true"
                data-nimg="1"
                decoding="async"
                height="12"
                loading="lazy"
                src="/assets/images/sidebar-item-icon.svg"
                style="color: transparent;"
                width="18"
              />
              <p
                class="MuiTypography-root MuiTypography-bodySmall css-fmwgzn-MuiTypography-root"
                title="Test-3"
              >
                Test-3
              </p>
            </a>
          </ul>
          <div
            aria-label="Video tags"
            class="MuiBox-root css-1gpn19b-TagsContainer-root"
            data-testid="sidebar-tags"
            role="list"
          >
            <a
              aria-pressed="false"
              class="MuiButtonBase-root MuiChip-root MuiChip-outlined MuiChip-sizeSmall MuiChip-colorDefault MuiChip-clickable MuiChip-clickableColorDefault MuiChip-outlinedDefault css-mtehtf-MuiButtonBase-root-MuiChip-root"
              data-testid="sidebar-tags-Test-1"
              href="/videos?tag=Test-1"
              role="button"
              tabindex="0"
            >
              <span
                class="MuiChip-label MuiChip-labelSmall css-19m61dl-MuiChip-label"
              >
                #Test-1
              </span>
            </a>
            <a
              aria-pressed="false"
              class="MuiButtonBase-root MuiChip-root MuiChip-outlined MuiChip-sizeSmall MuiChip-colorDefault MuiChip-clickable MuiChip-clickableColorDefault MuiChip-outlinedDefault css-mtehtf-MuiButtonBase-root-MuiChip-root"
              data-testid="sidebar-tags-Test-2"
              href="/videos?tag=Test-2"
              role="button"
              tabindex="0"
            >
              <span
                class="MuiChip-label MuiChip-labelSmall css-19m61dl-MuiChip-label"
              >
                #Test-2
              </span>
            </a>
            <a
              aria-pressed="false"
              class="MuiButtonBase-root MuiChip-root MuiChip-outlined MuiChip-sizeSmall MuiChip-colorDefault MuiChip-clickable MuiChip-clickableColorDefault MuiChip-outlinedDefault css-mtehtf-MuiButtonBase-root-MuiChip-root"
              data-testid="sidebar-tags-Test-3"
              href="/videos?tag=Test-3"
              role="button"
              tabindex="0"
            >
              <span
                class="MuiChip-label MuiChip-labelSmall css-19m61dl-MuiChip-label"
              >
                #Test-3
              </span>
            </a>
          </div>
        </div>
      </nav>
    </div>
    <div
      class="MuiContainer-root MuiContainer-maxWidthXl css-11shws3-MuiContainer-root"
    >
      <div
        class="MuiBox-root css-17i13ng"
      >
        <div
          class="MuiBox-root css-uwwqev"
        >
          <div>
            Test Content
          </div>
        </div>
      </div>
      <div
        class="MuiBox-root css-1b115a6"
        data-testid="right-sidebar"
      >
        <div>
          Right Content
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
